### PR TITLE
Better prompt with pwd

### PR DESCRIPTION
**Pull request by xihuai18**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
         "requests~=2.28.2",
         "rich==13.3.1",
         "distro~=1.8.0",
+        "psutil",
```
Comment (TheR1D) [inline setup.py:13]: This should be installed only for Windows users.

File: sgpt/app.py [hunk 2]
```diff
@@ -12,17 +12,17 @@
 
 
 import os
-
 # To allow users to use arrow keys in the REPL.
 import readline  # pylint: disable=unused-import
 
+import psutil
```
Comment (TheR1D) [inline sgpt/app.py:18]: Should be imported only for Windows users.

File: sgpt/app.py [hunk 3]
```diff
@@ -128,8 +128,13 @@ def main(  # pylint: disable=too-many-arguments
...
         )
 
+
     if not code and shell and typer.confirm("Execute shell command?"):
-        os.system(full_completion)
+        # Get the parent process name.
+        shell_name = get_shell_name()
+
+        # print(f'executing {shell_name} -c "{full_completion}"')
+        subprocess.run([shell_name, "-c", full_completion], check=False)
```
Comment (TheR1D) [inline sgpt/app.py:137]: Change it for REPL mode as well.

File: sgpt/make_prompt.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
 import platform
-from os import getenv
-from os.path import basename, splitext
 from distro import name as distro_name
+from sgpt.utils import get_shell_name
 
 
 SHELL_PROMPT = """###
 Provide only {shell} commands for {os} without any description.
+The current working directory is {pwd}.
```
Comment (TheR1D) [inline sgpt/make_prompt.py:9]: Not sure about this change. Could you provide some use cases for it?

File: .gitignore
Comment (TheR1D) [inline .gitignore:1]: Using "personal" .gitignore solves this.
```
git config --global core.excludesfile $HOME/.gitignore
```

File: sgpt/utils.py [hunk 5]
```diff
@@ -19,6 +20,18 @@ def get_mode(cls, shell, code) -> "CompletionModes":
...
 
+def get_shell_name() -> str:
+    """
+    Get the name of the shell.
+
+    :return: Shell name.
+    """
+    proc = psutil.Process(os.getppid())
+    return next(
+        p.name() for p in proc.parents() if "sh" in p.name() or "cmd" in p.name()
```
Comment (TheR1D) [inline sgpt/utils.py:31]: This will not work with `zsh` or `bash` or other non `sh` shells, it will also crash sgpt.

**Comment by TheR1D** [comment]
Thank you for the PR.
Currently I don't think this is in some workable shape, since it will be crashing every user with non `sh` shells. Please have a look on `tests/integration_tests.py`, and add testing cases for the specifc feature you are implementing.

File: .gitignore
Comment (xihuai18) [reply .gitignore:1]: Learned! thanks

File: sgpt/utils.py (hunk 5 above)
Comment (xihuai18) [reply sgpt/utils.py:31]: I am not sure but it works in my zsh shell, the os is Ubuntu.
image

Comment (TheR1D) [reply sgpt/utils.py:31]: Ubuntu:
```
proc = psutil.Process(pid=459, name='bash', started='19:24:23')
AttributeError: 'Process' object has no attribute 'parents'
```

MacOS:
```
proc = psutil.Process(pid=1473, name='zsh', status='running', started='01:54:35')
StopIteration
```

Comment (xihuai18) [reply sgpt/utils.py]: Okay, I think this may results from that I did add the current proc itself into the list, and I fix it now.

**Pull request closed without merging**